src/build/stable/commands/install_global_dependencies/index.ts

```diff
@@ -0,0 +1,119 @@
+import { IOType } from 'child_process';
+import { readFile } from 'fs/promises';
+import { join } from 'path';
+
+import { execSyncPretty } from '../../utils/exec_sync_pretty';
+import { AZLE_PACKAGE_PATH } from '../../utils/global_paths';
+
+type Versions = {
+    [key in ValidFlag]: string;
+};
+
+type ValidFlag = 'node' | 'dfx' | 'rustc' | 'wasi2ic';
+
```

Comment:
For context this things used to have name (which is for display purposes, maybe we don't need it), script (which we might be able to just assume as per the comment below) and package (which was the name as it would appear in the package.json) but since the package was the same as they key for all of them expect rust I decided to just use rustc everywhere. But this whole thing could be as simple as an array of valid flag names if we wanted